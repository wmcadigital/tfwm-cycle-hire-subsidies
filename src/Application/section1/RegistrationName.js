import { useFormState } from "react-final-form";

import { required, name, composeValidators } from "../../common/validation";
import FormSection from "../../common/FormSection";
import Question from "../../common/Question";
import ProgressIndicator from "../../common/ProgressIndicator";
import TextInput from "../../common/TextInput";

const RegistrationName = () => {
  const stateApi = useFormState();

  const errorFirstName =
    stateApi.submitFailed && stateApi.hasValidationErrors
      ? stateApi.errors?.formData.FirstName
      : null;

  const errorLastName =
    stateApi.submitFailed && stateApi.hasValidationErrors
      ? stateApi.errors?.formData.LastName
      : null;

  const question = "What is your name";
  
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 1 of 2"
        sectionName="About you"
      />
      <Question text={question} isRequired={true} />
      <TextInput
        fieldName="formData.FirstName"
        label="First Name"
        validation={composeValidators(required, name)}
        error={errorFirstName}
        isRequired={true}
      />
      <TextInput
        fieldName="formData.LastName"
        label="Last Name"
        validation={composeValidators(required, name)}
        error={errorLastName}
        isRequired={true}
      />
    </FormSection>
  );
};

export default RegistrationName;
