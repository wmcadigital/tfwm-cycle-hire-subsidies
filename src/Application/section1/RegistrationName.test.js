import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import RegistrationName from "./RegistrationName";

jest.mock("react-final-form");

describe("RegistrationName", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<RegistrationName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "no",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<RegistrationName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
        firstName: "13545",
      },
      submitFailed: true,
      errors: { firstName: "Invalid", lastName: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<RegistrationName />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
