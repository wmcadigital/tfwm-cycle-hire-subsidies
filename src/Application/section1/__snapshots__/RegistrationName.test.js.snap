// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationName individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error={null}
    fieldName="formData.FirstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="formData.LastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`RegistrationName individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    error={null}
    fieldName="formData.FirstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    error={null}
    fieldName="formData.LastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;

exports[`RegistrationName renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your name?"
  />
  <TextInput
    fieldName="formData.FirstName"
    label="First Name"
    validation={[Function]}
  />
  <TextInput
    fieldName="formData.LastName"
    label="Last Name"
    validation={[Function]}
  />
</FormSection>
`;
